From: Wei Yang <richard.weiyang@gmail.com>
Subject: maple_tree: calculate new_end when needed
Date: Wed, 16 Oct 2024 00:22:20 +0000

Patch series "Following cleanup after introduce mas_wr_store_type()".

Here are two cleanups after introduction of mas_wr_store_type().
Patch 1 postpones new_end calculation when needed.
Patch 2 removes an unnecessary sanity check in mas_wr_slot_store().


This patch (of 2):

For wr_exact_fit/wr_new_root, we don't need to calculate new_end.

Let's postpone it until necessary.

Link: https://lkml.kernel.org/r/20241016002221.32428-1-richard.weiyang@gmail.com
Link: https://lkml.kernel.org/r/20241016002221.32428-2-richard.weiyang@gmail.com
Signed-off-by: Wei Yang <richard.weiyang@gmail.com>
Reviewed-by: Sidhartha Kumar <sidhartha.kumar@oracle.com>
Reviewed-by: Liam R. Howlett <Liam.Howlett@Oracle.com>
Cc: Lorenzo Stoakes <lorenzo.stoakes@oracle.com>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 lib/maple_tree.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/lib/maple_tree.c~maple_tree-calculate-new_end-when-needed
+++ a/lib/maple_tree.c
@@ -4202,13 +4202,13 @@ static inline enum store_type mas_wr_sto
 	if (!wr_mas->entry)
 		mas_wr_extend_null(wr_mas);
 
-	new_end = mas_wr_new_end(wr_mas);
 	if ((wr_mas->r_min == mas->index) && (wr_mas->r_max == mas->last))
 		return wr_exact_fit;
 
 	if (unlikely(!mas->index && mas->last == ULONG_MAX))
 		return wr_new_root;
 
+	new_end = mas_wr_new_end(wr_mas);
 	/* Potential spanning rebalance collapsing a node */
 	if (new_end < mt_min_slots[wr_mas->type]) {
 		if (!mte_is_root(mas->node) && !(mas->mas_flags & MA_STATE_BULK))
_
